libass/ass_fontselect.c

```diff
@@ -601,6 +599,12 @@ find_font(ASS_FontSelector *priv, ASS_Library *library,
             break;
     }
 
+    if (force_full_match && score_min != 0)
+        return NULL;
+
+    if (selected != NULL)
+        *name_match = true;
```

Comment:
Is it right that this is moved here, after the `check_glyph`?

Doesn’t this mean that if the named font doesn’t contain the desired glyph, it will be re-added via `match_fonts` every single time when this glyph attempts to be rendered? That seems like it will both be slow and keep consuming infinitely more memory.

How about reverting the `*name_match` assignments and moving the `force_full_match` check inside the `matches_family_name` branch or the inner `for` loop as `if (force_full_match && score) continue;`?